src/include/duckdb/function/table_function.hpp

```diff
@@ -294,6 +294,10 @@ class TableFunction : public SimpleNamedParameterFunction { // NOLINT: work-arou
 	//! Whether or not the table function can immediately prune out filter columns that are unused in the remainder of
 	//! the query plan, e.g., "SELECT i FROM tbl WHERE j = 42;" - j does not need to leave the table function at all
 	bool filter_prune;
+	//! Whether or not the table function supports sampling pushdown. If not supported a sample will be taken after the
+	//! table function.
+	bool sampling_pushdown = false;
```

Comment:
We cannot push this into the `TableFunction` itself - that should not be modified in this manner. Sampling should instead be pushed in the same manner as table filters are - and passed into the `init` (i.e. part of `TableFunctionInitInput`).